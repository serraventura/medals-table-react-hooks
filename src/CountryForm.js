import React from "react";
import { Form, Input, ColumnForm } from "./styledComponents";

const CountryForm = ({ onEnter }) => (
  <Form>
    <ColumnForm>
      <Input
        data-test="add-country-input"
        type="text"
        placeholder="add country name and hit enter"
        onKeyDown={e => {
          if (e.keyCode === 13) onEnter(e.target.value);
        }}
      />
    </ColumnForm>
  </Form>
);

export default CountryForm;
