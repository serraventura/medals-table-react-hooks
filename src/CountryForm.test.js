import React from "react";
import { shallow, mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import CountryForm from "./CountryForm";

configure({ adapter: new Adapter() });

let onEnter;

describe("<CountryForm />", () => {
  beforeEach(() => {
    onEnter = jest.fn();
  });

  it("should render without crashing", () => {
    const wrapper = shallow(<CountryForm />);
    expect(wrapper).toBeTruthy();
  });

  it("should match snapshot", () => {
    const wrapper = mount(<CountryForm />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle onEnter prop", () => {
    const wrapper = shallow(<CountryForm onEnter={onEnter} />);

    wrapper
      .find('*[data-test="add-country-input"]')
      .first()
      .simulate("keydown", { keyCode: 13, target: { value: "uk" } });

    expect(onEnter).toHaveBeenCalled();
    expect(onEnter).toHaveBeenCalledWith("uk");
  });
});
