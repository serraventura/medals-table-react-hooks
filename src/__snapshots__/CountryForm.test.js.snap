// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryForm /> should match snapshot 1`] = `ReactWrapper {}`;
